code/game/objects/items.dm

```diff
@@ -244,6 +244,11 @@
 	if(flags_item & DELONDROP)
 		qdel(src)
 
+///Called whenever an item is unequipped to a new loc (IE, not when the item ends up in the hands)
+/obj/item/proc/removed_from_inventory(mob/user)
+	SEND_SIGNAL(src, COMSIG_ITEM_REMOVED_INVENTORY, user)
+	return
```

Comment:
For the record adding new signals is fine as long as they're in generic procs but try not to overdo it